pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```
@@ -0,0 +1,319 @@
+
+        SecureBatchReader reader = new SecureBatchReader(scope, stream, controllerURI,
+                truststorePath, validateHostname,
+                username, password);
+        reader.read();
+    }
+
+    private static Options getOptions() {
+        final Options options = new Options();
+        options.addOption("s", "scope", true, "The scope name of the stream to read from.");
+        options.addOption("n", "stream", true, "The name of the stream to read from.");
+        options.addOption("u", "uri", true, "The URI to the controller in the form tls://host:port");
+        options.addOption("t", "truststore", true,
+                "The location of .pem truststore file in the file system to use by this application process.");
+        options.addOption("v", "validatehost", false, "Whether to verify server's hostname.");
+        options.addOption("a", "username", true,
+                "The account username to use by the client for authenticating to the server.");
+        options.addOption("p", "password", true,
+                "The account password to use by the client for authenticating to the server.");
+        return options;
+    }
+
+    private static CommandLine parseCommandLineArgs(Options options, String[] args) throws ParseException {
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = parser.parse(options, args);
+        return cmd;
+    }
+
+    public void run() {
+        LOG.info("Starting segment logger");
+        this.executor = newScheduledThreadPool(THREAD_POOL_SIZE, String.format("EventsLogger"));
+        try {
+            Futures.delayedTask(() -> {
+                logEventCounts();
+                this.executor.shutdownNow();
+                return null;
+            }, Duration.ofMinutes(0), executor).get();
+        } catch (InterruptedException | ExecutionException e) {
+           e.printStackTrace();
```

Let the exception bubble up.

---

Yes Andre